pkg/wallet/internal/mapping/mapping_list_actions.go

```
@@ -0,0 +1,187 @@
+		result.LabelQueryMode = &labelQueryMode
+	}
+
+	if args.SeekPermission != nil {
+		seekPermission := primitives.BooleanDefaultTrue(*args.SeekPermission)
+		result.SeekPermissions = &seekPermission
+	}
+
+	if args.IncludeInputs != nil {
+		includeInputs := primitives.BooleanDefaultFalse(*args.IncludeInputs)
+		result.IncludeInputs = &includeInputs
+	}
+
+	if args.IncludeOutputs != nil {
+		includeOutputs := primitives.BooleanDefaultFalse(*args.IncludeOutputs)
+		result.IncludeOutputs = &includeOutputs
+	}
+
+	if args.IncludeLabels != nil {
+		includeLabels := primitives.BooleanDefaultFalse(*args.IncludeLabels)
+		result.IncludeLabels = &includeLabels
+	}
+
+	if args.IncludeInputSourceLockingScripts != nil {
+		includeInputSourceLockingScripts := primitives.BooleanDefaultFalse(*args.IncludeInputSourceLockingScripts)
+		result.IncludeInputSourceLockingScripts = &includeInputSourceLockingScripts
+	}
+
+	if args.IncludeInputUnlockingScripts != nil {
+		includeInputUnlockingScripts := primitives.BooleanDefaultFalse(*args.IncludeInputUnlockingScripts)
+		result.IncludeInputUnlockingScripts = &includeInputUnlockingScripts
+	}
+
+	if args.IncludeOutputLockingScripts != nil {
+		includeOutputLockingScripts := primitives.BooleanDefaultFalse(*args.IncludeOutputLockingScripts)
+		result.IncludeOutputLockingScripts = &includeOutputLockingScripts
+	}
+
+	return result
+}
+
+// MapListActionsResult maps *wdk.ListActionsResult to *sdk.ListActionsResult
+func MapListActionsResult(result *wdk.ListActionsResult) *sdk.ListActionsResult {
+	return &sdk.ListActionsResult{
+		TotalActions: result.TotalActions,
+		Actions:      slices.Map(result.Actions, mapListActionsAction),
+	}
+}
+
+// mapListActionsAction maps wdk.WalletAction to sdk.Action
+func mapListActionsAction(action wdk.WalletAction) sdk.Action {
+	var txidHash chainhash.Hash
+	if action.TxID != "" {
+		if hash, err := chainhash.NewHashFromHex(action.TxID); err == nil {
```

I would return an error if there is any, instead of swallowing it